### PR TITLE
Support JSON

### PR DESCRIPTION
**Pull request by sazzad16**
Reference #143

### CODE REVIEW

File: src/test/java/io/redisearch/CreateTest.java [hunk 1]
```diff
@@ -60,7 +50,7 @@ public void allIndexDefinition() throws Exception {
     indexRule.setPrefixes("person:");
     indexRule.setScore(0.818656);
     indexRule.setScoreFiled("myScore");
-    indexRule.setType(IndexDefinition.Type.HASH);
+    // indexRule.setType(IndexDefinition.Type.HASH); // Deprecated; moved in constructor
```
Comment (gkorland) [inline src/test/java/io/redisearch/CreateTest.java:53]: @sazzad16 why do you leave this commented line?

File: src/test/java/io/redisearch/client/ClientJsonTest.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+        @Override
+        public byte[] getRaw() {
+            return ("JSON." + name()).getBytes();
+        }
+    }
+
+    private static final String INDEX_NAME = "json-index";
+
+    private static Client search;
```
Comment (gkorland) [inline src/test/java/io/redisearch/client/ClientJsonTest.java:31]: @sazzad16 why is it static?

File: src/main/java/io/redisearch/client/IndexDefinition.java [hunk 3]
```diff
@@ -5,150 +5,167 @@
...
+    private String[] prefixes;
+    private String filter;
+    private String languageField;
+    private String language;
+    private String scoreFiled;
+    private double score = 1.0; // Default score when score isn't defined
+    private String payloadField;
+
+    public IndexDefinition() {
+        this(null);
```
Comment (gkorland) [inline src/main/java/io/redisearch/client/IndexDefinition.java:29]: @sazzad16 why null? and not keeping the HASH as default?

File: src/test/java/io/redisearch/CreateTest.java [hunk 4]
```diff
@@ -34,23 +34,13 @@ public void allOptions() throws Exception {
     defaultOptions.serializeRedisArgs(arrayList);
 
     assertEquals(
-        Arrays.asList("ON", "HASH", "NOOFFSETS", "NOFIELDS", "NOFREQS", "TEMPORARY", "1234", "STOPWORDS", "2", "stop", "run"),
+        Arrays.asList("NOOFFSETS", "NOFIELDS", "NOFREQS", "TEMPORARY", "1234", "STOPWORDS", "2", "stop", "run"),
         arrayList);
   }
 
-  @Test
```
Comment (gkorland) [inline src/test/java/io/redisearch/CreateTest.java:41]: @sazzad16 why did you delete the default empty test?

**Review by gkorland — CHANGES_REQUESTED** [review]

File: src/test/java/io/redisearch/CreateTest.java (hunk 4 above)
Comment (sazzad16) [reply src/test/java/io/redisearch/CreateTest.java:41]: @gkorland This test checking if `ON HASH` is being added in FT.CREATE command arguments. But I'm changing this behavior in this PR.

RediSearch accepts and processes FT.CREATE command without ON option. So empty constructor should not send any ON option to RediSearch, rather should depend on RediSearch's default behavior. This is what I'm doing in this PR and as a result this test won't work.

Note-1: Currently, RediSearch works exactly same as without ON option and with ON HASH.

Note-2: This change does not change any client behavior.

File: src/main/java/io/redisearch/client/IndexDefinition.java (hunk 3 above)
Comment (sazzad16) [reply src/main/java/io/redisearch/client/IndexDefinition.java:29]: @gkorland By null, we're using the default behavior of RediSearch. By using HASH, JRediSearch is defining its own default behavior, ignoring RediSearch's behavior.
https://github.com/RediSearch/JRediSearch/pull/145/files#r656079493

File: src/test/java/io/redisearch/client/ClientJsonTest.java (hunk 2 above)
Comment (sazzad16) [reply src/test/java/io/redisearch/client/ClientJsonTest.java:32]: @gkorland For doing search.close() in tearDown().

**Review by gkorland — CHANGES_REQUESTED** [review]

**Review by gkorland — APPROVED** [review]
